### PR TITLE
doc: Document mod_acl_user_groups

### LINKED ISSUE #1240: Document ACL user groups and content groups
Particularly if we want to replace other ACL modules with user acl_user_groups: #1131.

### PR DESCRIPTION
**Pull request by ddeboer**
Fix #1240.

### CODE REVIEW

**Comment by ArthurClemens** [comment]
I was anticipating this documentation, highly appreciated.

I found 2 missing pieces:

* Line 55 ends halfway
* Line 100 lacks the rule example

**Comment by ddeboer** [comment]
Yeah, it’s still WIP = work in progress. :wink:

By the way, if you wish to comment on individual lines, it’s easier to do so on the ‘Files changed’ tab.

File: doc/ref/modules/mod_acl_user_groups.rst [hunk 1]
```diff
@@ -1,31 +1,123 @@
...
+-----------------------
+
+By default, Zotonic has two content groups:
+
+* default (all site content)
+* system content (categories, predicates and user groups)
+
+To add or remove content groups, go to *Structure* > *Content groups* in the
+admin. Just like user groups, content groups are ordered in a hierarchy. So
+if you add a group under default, say ‘
```
Comment (ddeboer) [inline doc/ref/modules/mod_acl_user_groups.rst]: TODO: fixme

File: doc/ref/modules/mod_acl_user_groups.rst [hunk 2]
```diff
@@ -1,31 +1,123 @@
...
+admin, go to *Auth* > *Access control rules* > *Modules* tab to edit them.
+
+Deny rules
+----------
+
+By default, rules grant some permissions. But sometimes you want to deny some
+permissions that are granted by another rule. For instance, if you have a rule
+that allows anonymous users to view all content groups, but you have a special
+content group that you want to hide from anonymous users, add a rule to deny
+access:
```
Comment (ddeboer) [inline doc/ref/modules/mod_acl_user_groups.rst]: TODO: add example

**Comment by ddeboer** [comment]
Now ready for review!

**Comment by ddeboer** [comment]
Resolved conflicts by rebasing on master.

**Comment by mworrell** [comment]
Great!  👍

**Comment by mworrell** [comment]
Also merged into 0.x

**Pull request merged**